src/views/catalog/wizard/tabs/scripts/components/DynamicSSHKeyInjectionWizard.tsx

```diff
@@ -23,18 +27,18 @@ const DynamicSSHKeyInjectionWizard = () => {
 
   const onSubmit = (checked: boolean) => {
     updateVM((vmDraft) => {
-      const cloudInitConfigDrive = getVolumes(vm)?.find((v) => v.cloudInitConfigDrive);
+      const cloudInitVolume = getCloudInitVolume(vm);
 
-      if (cloudInitConfigDrive) {
+      if (cloudInitVolume) {
         vmDraft.spec.template.spec.volumes = [
-          ...getVolumes(vm).filter((v) => !v.cloudInitConfigDrive),
+          ...getVolumes(vm).filter((v) => !v.cloudInitConfigDrive || !v.cloudInitNoCloud),
```

Comment:
Maybe `!getCloudInitData(v)` could work here, too?

Reply:
That doesn't work I'm afraid.
!cond1 || !cond2 is not equivalant to !(cond1 || cond2)

Reply:
Yes, I do think that you need && there instead 👍🏽